runtime/src/cost_tracker.rs

```diff
@@ -97,10 +97,21 @@ impl CostTracker {
 
     pub fn update_execution_cost(
         &mut self,
-        _estimated_tx_cost: &TransactionCost,
-        _actual_execution_cost: u64,
+        estimated_tx_cost: &TransactionCost,
+        actual_execution_units: Option<u64>,
     ) {
-        // TODO: adjust block_cost / vote_cost / account_cost by (actual_execution_cost - execution_cost)
+        if let Some(actual_execution_units) = actual_execution_units {
+            let mut adjustment =
+                TransactionCost::new_with_capacity(estimated_tx_cost.writable_accounts.len());
+            adjustment
+                .writable_accounts
+                .extend(&estimated_tx_cost.writable_accounts);
```

Comment:
There should be a way to do this without copying `writable_accounts` by writing a special-case function for this adjustment.

Reply:
lower-level function can take an independent iterator and this can wrap it

Reply:
add function to specifically handle execution units update, avoid copying accounts hashmap